src/k8s-configuration/azext_k8s_configuration/_utils.py

```diff
@@ -0,0 +1,115 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import io
+import base64
+from azure.cli.core.azclierror import InvalidArgumentValueError, \
+    RequiredArgumentMissingError, MutuallyExclusiveArgumentError
+
+from paramiko.ed25519key import Ed25519Key
+from paramiko.ssh_exception import SSHException
+from Crypto.PublicKey import RSA, ECC, DSA
+
+
+def _get_protected_settings(ssh_private_key, ssh_private_key_file, https_user, https_key):
+    protected_settings = {}
+    ssh_private_key_data = _get_data_from_key_or_file(ssh_private_key, ssh_private_key_file)
+
+    # Add gitops private key data to protected settings if exists
+    # Dry-run all key types to determine if the private key is in a valid format
+    invalid_rsa_key, invalid_ecc_key, invalid_dsa_key, invalid_ed25519_key = (False, False, False, False)
+    if ssh_private_key_data != '':
+        try:
+            RSA.import_key(_from_base64(ssh_private_key_data))
+        except ValueError:
+            invalid_rsa_key = True
+        try:

```

Comment:
May be do this only if it was not a valid rsa key, and similarly check for invalid rsa key and invalid ecc key before importing of dsa key, etc.?

Reply:
Updated with fix